cmd/thanos/compact.go

```diff
@@ -200,6 +200,10 @@ func runCompact(
 		srv.Shutdown(err)
 	})
 
+	if err := configureGoAutoMemLimit(conf.goMemLimitConf); err != nil {
```

Comment:
Too bad we have to copy this everywhere. It would have been nice to only have it in main.go.

Reply:
Oh cool!
I didn't knew we can have that.
I'm move the code to be placed on the main.